modules/KoreBuild.Tasks/VerifyChecksum.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Build.Framework;
+
+namespace KoreBuild.Tasks
+{
+    /// <summary>
+    /// Verify the checksum for a single file.
+    /// </summary>
+    public class VerifyChecksum : Microsoft.Build.Utilities.Task
+    {
+        /// <summary>
+        /// The file path.
+        /// </summary>
+        [Required]
+        public string File { get; set; }
+
+        /// <summary>
+        /// The algorithm. Allowed values: SHA256, SHA384, SHA512.
+        /// </summary>
+        public string Algorithm { get; set; } = "SHA256";
```

Comment:
Should this be an Enum since we have a defined (and small) set of allowed values, or do tasks not play well with enums? Kind syntactic sugar either way.

Reply:
I wish. MSBuild doesn't support enum parameters on tasks.